stock_lot.py

```diff
@@ -31,20 +31,30 @@ def create_lot_quality_tests(cls, documents, template):
             test_to_save = []
             with Transaction().set_context(_check_access=False):
                 for lot in lots:
-                    used_template = None
-                    lot.active = False
-                    lot_to_save.append(lot)
-
-                    if not template:
-                        continue
-                    used_template = getattr(lot.product.template,
-                        template+'_quality_template')
-                    test = QualityTest(
-                        test_date=datetime.now(),
-                        templates=[used_template],
-                        document=str(lot))
-                    test.apply_template_values()
-                    test_to_save.append(test)
+                    if lot.quality_tests:
```

Comment:
You only need to check if there are quality controls, since the quality control workflow already takes care of activating or deactivating the lots.
So it would remain.


```
if lot.quality_test:
     contiune

used_template = None
lot.active = False
lot_to_save.append(lot)
￼...￼ 
```